src/main/java/io/aiven/kafka/auth/json/AivenAcl.java

```diff
@@ -38,26 +38,34 @@ public class AivenAcl {
     @SerializedName("resource_pattern")
     public final String resourceRePattern;
 
+    @SerializedName("permission_type")
+    private final AclPermissionType permissionType;
+
     public AivenAcl(final String principalType,
                     final String principal,
                     final String operation,
                     final String resource,
-                    final String resourcePattern) {
+                    final String resourcePattern,
+                    final AclPermissionType permissionType) {
         this.principalType = principalType;
         this.principalRe = Pattern.compile(principal);
         this.operationRe = Pattern.compile(operation);
-        this.resourceRe = Objects.nonNull(resource)
-            ? Pattern.compile(resource) : null;
+        this.resourceRe = Objects.nonNull(resource) ? Pattern.compile(resource) : null;
         this.resourceRePattern = resourcePattern;
+        this.permissionType = Objects.isNull(permissionType) ? AclPermissionType.ALLOW : permissionType;
+    }
+
+    public AclPermissionType getPermissionType() {
+        return Objects.nonNull(permissionType) ? permissionType : AclPermissionType.ALLOW;
```

Comment:
Given the initialization, seems `permissionType` is always not null. Maybe remove validation?

Reply:
This is a Gson deserialization issue. It does not call the constructor, so if you don't set `permission_type` in the json, it will remain null. I tried implementing a custom deserializer for `AclPermissionType`, but it's not called if you don't populate `permission_type` in the json. The only real solution would be to write a custom deserializer for `AivenAcl`, and it would also spare us from those weird checks that we do for null principal and operation, as they should never be null and the parsing shoudl fail instead.
However this needs more refactor and time, so for now I parked it and decided it to go for a private member and a getter which also deals with null.

Reply:
make sense. maybe just a comment to remind this would be enough